pkg/kubectl/cmd/util/config_flags.go

```diff
@@ -185,22 +181,12 @@ func (f *ConfigFlags) ToDiscoveryClient() (discovery.CachedDiscoveryInterface, e
 		cacheDir = *f.CacheDir
 	}
 
-	if len(cacheDir) > 0 {
-		wt := config.WrapTransport
-		config.WrapTransport = func(rt http.RoundTripper) http.RoundTripper {
-			if wt != nil {
-				rt = wt(rt)
-			}
-			return transport.NewCacheRoundTripper(cacheDir, rt)
-		}
-	}
-
 	discoveryClient, err := discovery.NewDiscoveryClientForConfig(config)
 	if err != nil {
 		return nil, err
 	}
 	cacheDir = computeDiscoverCacheDir(filepath.Join(homedir.HomeDir(), ".kube", "cache", "discovery"), config.Host)
-	return NewCachedDiscoveryClient(discoveryClient, cacheDir, time.Duration(10*time.Minute)), nil
+	return discovery.NewCachedDiscoveryClient(config.WrapTransport, discoveryClient, cacheDir, time.Duration(10*time.Minute)), nil
```

Comment:
Does this even do the same thing if the discoveryclient isn't constructed with the correct transport?  Seems doubtful.